detect_secrets/core/secrets_collection.py

```diff
@@ -200,7 +201,8 @@ def scan_file(self, filename, filename_key=None):
 
         if os.path.islink(filename):
             return False
-
+        if (os.path.splitext(filename)[1] in IGNORED_FILE_EXTENSIONS):
```

Comment:
(Just writing this down for posterity.)

We could fill two needs with one deed if we put this in `_results_accumulator` 
i.e.
```python
if os.path.splitext(filename)[1] in IGNORED_FILE_EXTENSIONS:
    return
```
This is because for regular `detect-secrets`, the control flow goes `scan_file`->`_extract_secrets_from_file`->`_results_accumulator`. And for `detect-secrets-server` the control flow goes `scan_diff`->`_results_accumulator`.

However, **let's leave the location as-is for this PR, it's better** 😀  👍 . The current location where we [filter out ignored file extensions](https://github.com/Yelp/detect-secrets-server/blob/1905e80acf8cdb6318ee5ad416a0e3724755f1c7/detect_secrets_server/storage/core/git.py#L96) in `detect-secrets-server`, is a lot earlier on, (before `scan_diff` ever gets called) and better performance-wise. So it's not worth it to make it super DRY at the cost of performance IMO.